src/main/kotlin/com/dansplugins/factionsystem/command/faction/claim/MfFactionClaimFillCommand.kt

```diff
@@ -89,6 +89,11 @@ class MfFactionClaimFillCommand(private val plugin: MedievalFactions) : CommandE
                             sender.sendMessage("$RED${plugin.language["CommandFactionClaimFillNoClaimableChunks"]}")
                             return@saveChunks
                         }
+                        val claimFillMaxChunks = plugin.config.getInt("factions.claimFillMaxChunks", -1)
+                        if (claimFillMaxChunks > 0 && claimableChunks.size > claimFillMaxChunks) {
```

Comment:
Putting the check here will incur a performance penalty, as the `fill` function will still try to find more chunks than the cap. This check should go in `fill` instead and apply a circuit break as soon as it finds itself trying to find more chunks than the limit.

Reply:
This check has been moved to the `fill` function. An exception is now thrown if it finds itself trying to find more chunks than the limit.